mail_multicompany/models/mail_message.py

```diff
@@ -19,14 +19,20 @@ def create(self, vals_list):
             if not vals.get("company_id"):
                 vals["company_id"] = self.env.company.id
             if not vals.get("mail_server_id"):
-                vals["mail_server_id"] = (
-                    self.sudo()
-                    .env["ir.mail_server"]
-                    .search(
-                        [("company_id", "=", vals.get("company_id", False))],
-                        order="sequence",
-                        limit=1,
-                    )
-                    .id
+                mail_server, email_from = self._find_company_mail_server(
+                    vals.get("email_from", False), vals.get("company_id", False)
                 )
+                vals["mail_server_id"] = mail_server.id if mail_server else False
+                vals["email_from"] = email_from
```

Comment:
instead of:

```
vals["key1"] = "val1"
vals["key2"] = "val2"
...
```

It's better to:

```
vals.update({
    "key1": "val1",
    "key2": "val2",
    ...
})
```

As a single update would be faster.

Reply:
You're right, thanks.